src/main/java/seedu/address/model/Model.java

```diff
@@ -245,12 +243,12 @@
      * Updates the filter of the filtered module list to filter by the given {@code predicate}.
      * @throws NullPointerException if {@code predicate} is null.
      */
-    void updateFilteredModuleList(TitleContainsKeywordsPredicate predicate);
+    void updateFilteredModuleList(Predicate<Module> predicate);
```

Comment:
is there a reason to not use TitleContainsKeywordsPredicate

Reply:
TitleContainsKeywordPredicate implements Predicate. Therefore in terms of LSP, having it as Predicate would be more accurate.